src/keyboard/Key.tsx

```diff
@@ -53,35 +67,34 @@ export const Key = ({
 }: PropsWithChildren<KeyProps>) => {
   const size = makeSize(props, oneU);
 
-  const children = Children.map(props.children, (c) => (
-    <div
-      data-zoomer={hoverZoom}
-      className="justify-self-center self-center row-start-2 row-end-3 col-start-2 col-end-3 font-keycap text-lg data-[zoomer=true]:group-hover:text-2xl"
-    >
-      {c}
-    </div>
-  ));
-
   return (
     <div
-      className="group inline-flex b-0 justify-content-center items-center transition-all duration-100 data-[zoomer=true]:hover:translate-y-[calc(-1em-7px)] data-[zoomer=true]:hover:translate-x-[calc(-1em)]"
+      className="group justify-content-center group items-center select-none transition-transform origin-center data-[zoomer=true]:hover:scale-[2] data-[zoomer=true]:hover:z-20"
       data-zoomer={hoverZoom}
       style={size}
       {...props}
     >
       <button
         aria-selected={selected}
         data-zoomer={hoverZoom}
-        className={`rounded${
-          oneU > 20 ? "-md" : ""
-        } transition-all duration-100 m-auto p-0 b-0 box-border grid grid-rows-[0_var(--zmk-key-center-height)_0] grid-cols-[0_var(--zmk-key-center-width)_0] data-[zoomer=true]:hover:grid-rows-[1em_var(--zmk-key-center-height)_1em] data-[zoomer=true]:hover:grid-cols-[1em_var(--zmk-key-center-width)_1em] shadow-[0_0_0_1px_inset] shadow-base-content data-[zoomer=true]:shadow-base-200 data-[zoomer=true]:hover:shadow-base-content data-[zoomer=true]:hover:z-50 text-base-content bg-base-100 aria-selected:bg-primary aria-selected:text-primary-content grow @container`}
+        className={[
+          oneU > 20 ? "rounded-md" : "rounded",
+          "relative w-[var(--zmk-key-center-width)] h-[var(--zmk-key-center-height)] bg-base-100 border border-transparent",
+          "group-hover:border-[#a6adbb] aria-selected:bg-primary aria-selected:text-primary-content @container",
+        ].join(' ')}
       >
         {header && (
-          <span className="p-0 b-0 m-0 text-xs w-full h-full text-nowrap justify-self-start row-start-1 row-end-2 col-start-1 col-end-4 hidden group-hover:inline-block group-hover:truncate @md:underline">
+          <span className="opacity-80 truncate hidden group-hover:block text-[6px] text-center uppercase absolute top-1 left-[4px] right-[4px] max-w-[90%] h-4 leading-none">
```

Comment:
more hard coded values here, can we avoid them?

Reply:
absolutely, i'll revise and address the others as well.